src/query/block/container.go

```diff
@@ -22,38 +22,71 @@ package block
 
 import (
 	"errors"
+	"fmt"
 	"time"
 
 	"github.com/m3db/m3/src/query/ts"
 	xerrors "github.com/m3db/m3/src/x/errors"
 )
 
 var (
-	errMismatchedStepIter   = errors.New("container step iter has mismatched step size")
-	errMismatchedUcStepIter = errors.New("unconsolidated container step iter has mismatched step size")
+	errMismatchedStepIter = errors.New(
+		"container step iter has mismatched step size")
+	errMismatchedUcStepIter = errors.New(
+		"unconsolidated container step iter has mismatched step size")
+	errNoContainerBlocks = errors.New(
+		"no blocks provided to initialize container block")
 )
 
 type containerBlock struct {
 	err    error
+	meta   Metadata
 	blocks []Block
 }
 
-func newContainerBlock(blocks []Block) AccumulatorBlock {
+func newContainerBlock(blocks []Block) (AccumulatorBlock, error) {
+	if len(blocks) == 0 {
+		return nil, errNoContainerBlocks
+	}
+
+	meta := blocks[0].Meta()
+	for _, b := range blocks[1:] {
+		m := b.Meta()
+		// FIXME ARNIKOLA use actual equality here
```

Comment:
??

Reply:
Bad comment, but waiting on https://github.com/m3db/m3/pull/1871 which adds an `Equals` function to the metas